src/charm.py

```diff
@@ -366,52 +211,16 @@ def _on_ingress_revoked(self, _event):
     def _on_list_receivers_action(self, event: ops.ActionEvent):
         res = {}
         for receiver in self._requested_receivers():
-            res[receiver.replace("_", "-")] = (
-                f"{self.ingress.external_host or self.tempo.url}:{self.tempo.receiver_ports[receiver]}"
-            )
+            res[receiver.replace("_", "-")] = self.get_receiver_url(receiver)
         event.set_results(res)
 
     # keep this event handler at the bottom
-    def _on_collect_unit_status(self, e: CollectStatusEvent):
-        # todo add [nginx.workload] statuses
-
-        if not self.tempo.is_ready:
-            e.add_status(WaitingStatus("[workload.tempo] Tempo API not ready just yet..."))
-
-        # TODO: should we set these statuses on the leader only, or on all units?
-        if issues := self._inconsistencies:
-            for issue in issues:
-                e.add_status(BlockedStatus("[consistency.issues]" + issue))
-            e.add_status(BlockedStatus("[consistency] Unit *disabled*."))
-        else:
-            if self.is_clustered:
-                # no issues: tempo is consistent
-                if not self.coordinator.is_recommended:
-                    e.add_status(ActiveStatus("[coordinator] degraded"))
-                else:
-                    e.add_status(ActiveStatus())
-            else:
-                e.add_status(ActiveStatus())
-
-    def _on_nginx_pebble_ready(self, _) -> None:
-        self.nginx.configure_pebble_layer()
-
-    def _on_nginx_prometheus_exporter_pebble_ready(self, _) -> None:
-        self.nginx_prometheus_exporter.configure_pebble_layer()
-
-    def _on_event(self, event) -> None:
-        """A set of common configuration actions that should happen on every event."""
-        if isinstance(event, CollectStatusEvent):
-            return
-        # plan layers
-        self.nginx.configure_pebble_layer()
-        self.nginx_prometheus_exporter.configure_pebble_layer()
-        # configure ingress
-        self._configure_ingress()
-        # update cluster relations
-        self._update_tempo_cluster()
-        # update tracing relations
-        self._update_tracing_relations()
+    def _on_collect_unit_status(self, e: ops.CollectStatusEvent):
+        pass
+        # add Tempo charm custom blocking conditions
```

Comment:
is this something we want to do in a follow-up PR, or why are we leaving this TODO here?

I think the coordinator is happy if the existing consistency checks are working. IIRC the coordinator will already set blocked if the consistency check in `__init__` fails and the coordinator is not responsive. Perhaps that's enough and we don't need to add anything on top, in that case we can remove this event handler altogether.